### ISSUE TITLE
Passing non-serializable loss function since 4.0.0.

### ISSUE THREAD

**Issue by Deimos357** [microsoft/LightGBM#6056]
Hello. I have custom non-serializable loss function which I was passing through _fobj_ param of _lgb.train_.
Since https://github.com/microsoft/LightGBM/pull/5052 it's not possible any more.
Also it's not possible to pass non-serializable loss function as _params['fobj']_ value, because of multiple _copy.deepcopy(params)_ usages in _train_ function.
So what is the way to pass such loss function in 4.0.0?

**Comment by jameslamb** [comment]
Can you please share the function definition and a minimal, reproducible example showing how you're using it and the error you're getting?
